### PR TITLE
PLAT-22530: support captions in Simulive

### PR DESCRIPTION
**Pull request by Amirhalfon1**
get the entry's captionAssets and put it in the serveFlavor's json for the Simulive flow

### CODE REVIEW

File: alpha/apps/kaltura/lib/myPlaylistUtils.class.php [hunk 1]
```diff
@@ -1427,15 +1427,15 @@ public static function getPlaylistDataFromEntries($entries, $flavorParamsIds, $c
 		return array($entryIds, $durations, $mediaEntry, $captionFiles);
 	}
 
-	protected static function getCaptionFilesForEntryIds($entryIds, $captionLanguages)
+	public static function getCaptionFilesForEntryIds($entryIds, $captionLanguages, $allLanguages = false)
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/lib/myPlaylistUtils.class.php:1430]: No need to add another flag to the function, we can set that $captionLanguages is default to null, and if it is not set that we grab all langueges

File: alpha/apps/kaltura/lib/kSimuliveUtils.php [hunk 2]
```diff
@@ -38,9 +38,14 @@ public static function getSimuliveEventDetails(LiveEntry $entry, $time)
 		$startTime = $currentEvent->getCalculatedStartTime() * self::SECOND_IN_MILLISECONDS;
 		$durations[] = min($sourceEntry->getLengthInMsecs(), ($currentEvent->getCalculatedEndTime() * self::SECOND_IN_MILLISECONDS) - $startTime);
 		$endTime = $startTime + array_sum($durations);
+		$assets = array();
 		// getting the flavors from source entry
-		$flavors = assetPeer::retrieveReadyWebByEntryId($sourceEntry->getId());
-		return array($durations, $flavors, $startTime, $endTime, $dvrWindowMs);
+		$flavorAssets = assetPeer::retrieveReadyWebByEntryId($sourceEntry->getId());
+		$assets = array_merge($assets, $flavorAssets);
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/lib/kSimuliveUtils.php:44]: Why is this needed? asset is empty.
Why not just create asset out of the merge of flavour and captions?

File: alpha/apps/kaltura/modules/extwidget/actions/serveFlavorAction.class.php [hunk 3]
```diff
@@ -915,22 +920,35 @@ protected function getLiveParams()
...
+			list ($file_sync, $path, $sourceType) = kFileSyncUtils::getFileSyncServeFlavorFields($syncKey, $asset, self::getPreferredStorageProfileId(), self::getFallbackStorageProfileId());
 			if(!$path)
 			{
-				KalturaLog::debug('missing path for flavor ' . $flavor->getId() . ' version ' . $flavor->getVersion());
+				KalturaLog::debug('missing path for asset ' . $asset->getId() . ' version ' . $asset->getVersion());
+			}
+			$sequence = array('clips' => array(self::getClipData($path, $asset, $sourceType)));
+			if (is_callable(array($asset, 'getLanguage')) && $asset->getLanguage())
+			{
+				$languageCode = languageCodeManager::getLanguageCode($asset->getLanguage(), true);
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/modules/extwidget/actions/serveFlavorAction.class.php:941]: This is a wrong approach, the correct way is to add all asset tree the getLanguage function, where who ever did not implement it will return null, while others will follow this logic.

File: alpha/apps/kaltura/lib/myPlaylistUtils.class.php (hunk 1 above)
Comment (Amirhalfon1) [reply alpha/apps/kaltura/lib/myPlaylistUtils.class.php:1430]: I'm removing this change, it isn't needed anymore (was needed before and forgot to remove it)

**Comment by MosheMaorKaltura** [comment]
reviewed
